turbopuffer/query.py

```diff
@@ -37,50 +40,69 @@
 
 @dataclass
 class VectorQuery:
-    vector: Optional[List[float]] = None
+    rank_by: RankInput
     distance_metric: Optional[str] = None
     top_k: int = 10
-    include_vectors: bool = False
     include_attributes: Optional[Union[List[str], bool]] = None
     filters: Optional[Filters] = None
-    rank_by: Optional[RankInput] = None
     consistency: Optional[ConsistencyDict] = None
-    vector_encoding_format: Optional[str] = None
+    vector_encoding: Optional[str] = None
 
     def from_dict(source: dict) -> 'VectorQuery':
         return VectorQuery(
-            vector=source.get('vector'),
             distance_metric=source.get('distance_metric'),
```

Comment:
let's remove this?

Reply:
Oh, funny, our docs have just been lying about this being required. Will do.

<img width="799" alt="image" src="https://github.com/user-attachments/assets/64168436-c952-4581-a60f-8c8a3f2e0a84" />

Reply:
we should require it in writes and remove it from queries entirely, so we can ensure the index is always built in advance